### PR TITLE
Exclude glassfish Executor which does not permit wrapped runnables

### LINKED ISSUE #589: The agent is unable to instrument ManagedExecutorService and crashes the thread
The APM agent is unable to instrument the `ManagedExecutorService`.
Whenever a `Callable` is submitted invocation crashes with:
```
SEVERE [org.glassfish.enterprise.concurrent] (EE-ManagedExecutorService-default-Thread-1) java.lang.ClassCastException: co.elastic.apm.agent.impl.async.ContextInScopeRunnableWrapper cannot be cast to org.glassfish.enterprise.concurrent.internal.ManagedFutureTask
```
**Environment**
OpenJDK 1.8.0_191 (build 1.8.0_191-8u191-b12-2ubuntu0.18.04.1-b12)
Wildfly 9.0.2.Final
APM agent 1.5.0, 1.6.0-SNAPSHOT at fb3c7cd5f15ee19f2c6d4ed4d5a5a2378e83412e

Default Wildfly EE configuration:
<details><summary>Click to expand</summary>

```xml
<subsystem xmlns="urn:jboss:domain:ee:3.0">
    <spec-descriptor-property-replacement>false</spec-descriptor-property-replacement>
    <concurrent>
        <context-services>
            <context-service name="default" jndi-name="java:jboss/ee/concurrency/context/default" use-transaction-setup-provider="true"/>
        </context-services>
        <managed-thread-factories>
            <managed-thread-factory name="default" jndi-name="java:jboss/ee/concurrency/factory/default" context-service="default"/>
        </managed-thread-factories>
        <managed-executor-services>
            <managed-executor-service name="default" jndi-name="java:jboss/ee/concurrency/executor/default" context-service="default" hung-task-threshold="60000" core-threads="5" max-threads="25" keepalive-time="5000"/>
        </managed-executor-services>
        <managed-scheduled-executor-services>
            <managed-scheduled-executor-service name="default" jndi-name="java:jboss/ee/concurrency/scheduler/default" context-service="default" hung-task-threshold="60000" core-threads="2" keepalive-time="3000"/>
        </managed-scheduled-executor-services>
    </concurrent>
    <default-bindings context-service="java:jboss/ee/concurrency/context/default" datasource="java:jboss/datasources/ExampleDS" jms-connection-factory="java:/activemq/ConnectionFactory" managed-executor-service="java:jboss/ee/concurrency/executor/default" managed-scheduled-executor-service="java:jboss/ee/concurrency/scheduler/default" managed-thread-factory="java:jboss/ee/concurrency/factory/default"/>
</subsystem>
```
</details>

Nothing to see in debug logs, except for loads of these:
```
ERROR [stderr] (ServerService Thread Pool -- 73) [ServerService Thread Pool -- 73] DEBUG co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers - co.elastic.apm.api.CaptureTransaction was loaded by org.jboss.modules.ModuleClassLoader
ERROR [stderr] (ServerService Thread Pool -- 73) [ServerService Thread Pool -- 73] DEBUG co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers - co.elastic.apm.api.CaptureSpan was loaded by org.jboss.modules.ModuleClassLoader
ERROR [stderr] (ServerService Thread Pool -- 73) [ServerService Thread Pool -- 73] DEBUG co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers - javax.ws.rs.Path was loaded by org.jboss.modules.ModuleClassLoader
ERROR [stderr] (ServerService Thread Pool -- 73) [ServerService Thread Pool -- 73] DEBUG co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers - javax.jws.WebService was loaded by org.jboss.modules.ModuleClassLoader
ERROR [stderr] (ServerService Thread Pool -- 73) [ServerService Thread Pool -- 73] DEBUG co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers - javax.faces.lifecycle.Lifecycle was loaded by org.jboss.modules.ModuleClassLoader
ERROR [stderr] (ServerService Thread Pool -- 73) [ServerService Thread Pool -- 73] DEBUG co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers - javax.faces.lifecycle.Lifecycle was loaded by org.jboss.modules.ModuleClassLoader
```

### PR DESCRIPTION
**Pull request by felixbarny**
fixes elastic/apm-agent-java#589

### CODE REVIEW

**Comment by felixbarny** [comment]
Contains no tests because we would have to instrument `java.util.concurrent.ThreadPoolExecutor` which is loaded from the bootstrap class loader and thus can't be instrumented in unit tests.

**Comment by felixbarny** [comment]
Snapshot for this PR: [elastic-apm-agent-1.6.1-SNAPSHOT.jar.zip](https://github.com/elastic/apm-agent-java/files/3090154/elastic-apm-agent-1.6.1-SNAPSHOT.jar.zip)

**Comment by felixbarny** [comment]
@mdindoffer could you give this snapshot a try?

[elastic-apm-agent-1.6.1-SNAPSHOT.jar.zip](https://github.com/elastic/apm-agent-java/files/3094753/elastic-apm-agent-1.6.1-SNAPSHOT.jar.zip)

**Comment by mdindoffer** [comment]
@felixbarny Perfect, that seems to do the trick.

File: apm-agent-plugins/apm-java-concurrent-plugin/src/main/java/co/elastic/apm/agent/concurrent/ExecutorInstrumentation.java [hunk 1]
```diff
@@ -61,21 +73,27 @@
     public ElementMatcher<? super TypeDescription> getTypeMatcher() {
         return hasSuperType(named("java.util.concurrent.Executor"))
             // hazelcast tries to serialize the Runnables/Callables to execute them on remote JVMs
-            .and(not(nameStartsWith("com.hazelcast")));
+            .and(not(nameStartsWith("com.hazelcast")))
+            .and(not(named("org.glassfish.enterprise.concurrent.internal.ManagedThreadPoolExecutor")));
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-java-concurrent-plugin/src/main/java/co/elastic/apm/agent/concurrent/ExecutorInstrumentation.java]: Do we need it here if it inherits both `submit` and `execute` methods?

Comment (felixbarny) [reply apm-agent-plugins/apm-java-concurrent-plugin/src/main/java/co/elastic/apm/agent/concurrent/ExecutorInstrumentation.java]: No, I forgot to remove it

**Pull request merged**